clouddriver-core/src/test/java/com/netflix/spinnaker/clouddriver/data/task/jedis/JedisTaskTest.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2020 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.data.task.jedis;
+
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.io.Resources;
+import com.netflix.spinnaker.clouddriver.data.task.DefaultTaskStatus;
+import com.netflix.spinnaker.clouddriver.data.task.TaskState;
+import java.nio.charset.StandardCharsets;
+import org.assertj.core.api.AssertionsForClassTypes;
+import org.junit.jupiter.api.Test;
+import org.mockito.InOrder;
+import org.mockito.Mockito;
+
+final class JedisTaskTest {
+  private static final ObjectMapper objectMapper = new ObjectMapper();
+  private static final String PHASE = "DEPLOY";
+
+  @Test
+  void serializationTest() throws Exception {
+    RedisTaskRepository taskRepository = mock(RedisTaskRepository.class);
+    JedisTask task = new JedisTask("123", 100, taskRepository, "owner", ImmutableList.of(), false);
+
+    DefaultTaskStatus oldStatus =
+        DefaultTaskStatus.create(PHASE, "Starting deploy", TaskState.STARTED);
+    DefaultTaskStatus status =
+        DefaultTaskStatus.create(PHASE, "Finished deploy", TaskState.COMPLETED);
+    Object results =
+        ImmutableMap.of("instances", ImmutableList.of("my-instance-v000", "my-instance-v001"));
+
+    when(taskRepository.getHistory(eq(task))).thenReturn(ImmutableList.of(oldStatus, status));
+    when(taskRepository.getResultObjects(eq(task))).thenReturn(ImmutableList.of(results));
+    when(taskRepository.currentState(eq(task))).thenReturn(status);
+
+    String result = objectMapper.writeValueAsString(task);
+    String expectedResult =
+        Resources.toString(JedisTaskTest.class.getResource("task.json"), StandardCharsets.UTF_8);
+
+    // Compare the parsed trees of the two results, which is agnostic to key order
+    AssertionsForClassTypes.assertThat(objectMapper.readTree(result))
```

Comment:
Using the normal `assertThat` doesn't work here as the result of `objectMapper.readTree(String)` is a `JsonNode`, which implements `Iterable`, so `AssertJ` tries to use the equality check for an iterable rather than the built-in `.equals` of `JsonNode`.

Reply:
You could also cast: `assertThat((Object) objectMapper.readTree(result))` but I guess that's no better really.

Reply:
Oh, I figured out where to reply to this...[AssertsionsForClassTypes](https://joel-costigliola.github.io/assertj/core-8/api/org/assertj/core/api/AssertionsForClassTypes.html) is the AssertJ recommended way of dealing with ambiguity in asserting on a class or one of its interfaces.